mne/io/fieldtrip/tests/test_fieldtrip.py

```diff
@@ -0,0 +1,206 @@
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info',
+                         all_test_params_epochs)
+def test_averaged(cur_system, version, use_info):
+    """Test comparing reading an Evoked object and the FieldTrip version."""
+    test_data_folder_ft = get_data_paths(cur_system)
+    mne_avg = get_averaged_data(cur_system)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'averaged_%s.mat' % (version,))
+    if version == 'v73' and not _has_h5py():
+        with pytest.raises(ImportError):
+            mne.io.read_evoked_fieldtrip(cur_fname, info)
+        return
+
+    with pytest.warns(**pytestwarning):
+        avg_ft = mne.io.read_evoked_fieldtrip(cur_fname, info)
+
+    mne_data = mne_avg.data[:, :-1]
+    ft_data = avg_ft.data
+
+    check_data(mne_data, ft_data, cur_system)
+    check_info_fields(mne_avg, avg_ft, use_info)
+
+
+@testing.requires_testing_data
+@pytest.mark.filterwarnings('ignore::RuntimeWarning')
+@pytest.mark.parametrize('cur_system, version, use_info',
+                         all_test_params_epochs)
+def test_epoched(cur_system, version, use_info):
+    """Test comparing reading an Epochs object and the FieldTrip version."""
+    has_pandas = _check_pandas_installed(strict=False) is not False
+    test_data_folder_ft = get_data_paths(cur_system)
+    mne_epoched = get_epoched_data(cur_system)
+    if use_info:
+        info = get_raw_info(cur_system)
+        pytestwarning = {'expected_warning': None}
+    else:
+        info = None
+        pytestwarning = no_info_warning
+
+    cur_fname = os.path.join(test_data_folder_ft,
+                             'epoched_%s.mat' % (version,))
+    if has_pandas:
+        pandas = _check_pandas_installed()
+        if version == 'v73' and not _has_h5py():
+            with pytest.raises(ImportError):
+                mne.io.read_epochs_fieldtrip(cur_fname, info)
+            return
+        with pytest.warns(**pytestwarning):
+            epoched_ft = mne.io.read_epochs_fieldtrip(cur_fname, info)
+        assert isinstance(epoched_ft.metadata, pandas.DataFrame)
+    else:
+        with pytest.warns(None) as warn_record:
+            if version == 'v73' and not _has_h5py():
+                with pytest.raises(ImportError):
+                    mne.io.read_epochs_fieldtrip(cur_fname, info)
+                return
+            epoched_ft = mne.io.read_epochs_fieldtrip(cur_fname, info)
+            assert epoched_ft.metadata is None
+            assert_warning_in_record(pandas_not_found_warning_msg, warn_record)
+            if pytestwarning['expected_warning'] is not None:
+                assert_warning_in_record(pytestwarning['message'], warn_record)
```

Comment:
the branches in the block above seems overly complicated. Multiple calls to `_has_h5py()`, return in the middle. Can you see if you can simplify?

Reply:
yes, this is a complicated test. there are, however, lots of different cases, this test should take into account:

1. is the `info` field of a raw fif file used or not. if not, check that the proper warning is emitted.
2. if we are working on a v7.3 matlab file and h5py is not installed, check for appropriate import exception.
3. if pandas is installed, check if event metadata is returned. if pandas is not installed, check for the proper warning.

the problem is that any combination of the three cases might occur and should be considered in the test. what further complicates the issue is that case 2 might throw an exception while the other two cases throw warnings. this needs to be considered as well.

i agree with you that the code should be simpler, but i do not have an idea how i can do it. so, if you have any suggestions, i would appreciate them.

ps: i could, of course, have a variable set to the output of `_has_h5py` and check for that value instead. if you insist, i will do it but i do not see how that would help the readability.